ios/BT/ExposureManager.swift

```diff
@@ -32,37 +32,41 @@ final class ExposureManager: NSObject {
     manager.invalidate()
   }
   
-  @objc func requestExposureNotificationAuthorization(_ callback: @escaping RCTResponseSenderBlock) {
-    manager.setExposureNotificationEnabled(true) { error in
-      if let error = error as? ENError, error.code == .notAuthorized {
-        callback([String.notAuthorized])
-      } else if let error = error {
-        callback([error.localizedDescription])
-      } else {
-        callback([String.authorized])
+  @objc func requestExposureNotificationAuthorization(authorized: Bool, callback: @escaping RCTResponseSenderBlock) {
+    // Ensure exposure notifications are enabled if the app is authorized. The app
+    // could get into a state where it is authorized, but exposure
+    // notifications are not enabled,  if the user initially denied Exposure Notifications
+    // during onboarding, but then flipped on the "COVID-19 Exposure Notifications" switch
+    // in Settings.
```

Comment:
This explains the four states in the enum then?

Reply:
```public enum ENAuthorizationStatus : Int {

    
    /// Authorization status has not yet been determined.
    	/// This status means the user has not been prompted yet. Using the API in this state may prompt the user.
    case unknown = 0

    
    /// This app is not authorized to use Exposure Notification. The user cannot change this app's authorization status.
    	/// This status may be due to active restrictions, such as parental controls being in place.
    case restricted = 1

    
    /// The user denied authorization for this app.
    case notAuthorized = 2

    
    /// The user has authorized this app to use Exposure Notification.
    case authorized = 3
}```
🧐